data/trie/sync.go

```diff
@@ -244,7 +273,16 @@ func (ts *trieSyncer) getNode(hash []byte) (node, error) {
 		return trieNode(n)
 	}
 
-	return nil, ErrNodeNotFound
+	existingNode, err := getNodeFromDBAndDecode(hash, ts.trie.Database(), ts.trie.marshalizer, ts.trie.hasher)
+	if err != nil {
+		return nil, ErrNodeNotFound
+	}
+	err = existingNode.setHash()
+	if err != nil {
+		return nil, ErrNodeNotFound
```

Comment:
ErrNodeNotFound ?

Reply:
deliberately returned as ErrNodeNotFound because that will trigger a request (for some reason the node's hash was not computed)

Reply:
ok